packages/dataviews/src/dataforms-layouts/data-form-layout.tsx

```diff
@@ -73,12 +103,53 @@ export function DataFormLayout< Item >( {
 					return children( FieldLayout, formField );
 				}
 
+				const errorMessage =
+					( fieldDefinition?.validationSchema?.onTouched &&
+						validation.touchedFields.includes(
+							fieldDefinition.id
+						) ) ||
+					! fieldDefinition?.validationSchema?.onTouched
+						? Object.values(
+								validation.errorMessages[ formField.id ] ?? []
+						  )[ 0 ]
+						: '';
+
 				return (
 					<FieldLayout
 						key={ formField.id }
 						data={ data }
 						field={ formField }
-						onChange={ onChange }
+						errorMessage={ errorMessage }
+						onChange={ ( value ) => {
+							onChange( value );
```

Comment:
Could we move the logic below to a helper function?
Also does this still work within a Panel layout given we pass in children there? https://github.com/WordPress/gutenberg/blob/3da37fdabdf27fff0344aaf7cb724f4e57b2768a/packages/dataviews/src/dataforms-layouts/panel/index.tsx#L159

Reply:
> Could we maybe also move this logic and trigger it in the DataForm component maybe?
The DataFormLayout is also being used within the panel layout, meaning every time the dropdown is opened this logic gets run again.

Good point, you're right! I will double-check this!

Reply:
> Just to clarify, for instance, you're referring to a field that can render max 5 characters in the input, right?

Yeah, or an email/link field. But you're right, it should be a UI thing. And thinking about it more, it would be kinda strange if someone triggered an update with the last "valid" state, although a user kept typing in an invalid state.